hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -84,12 +85,20 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
       // default parallelism
       int parallelism = dataStream.getExecutionConfig().getParallelism();
       DataStream<Object> pipeline;
-
-      // bootstrap
-      final DataStream<HoodieRecord> hoodieRecordDataStream =
-          Pipelines.bootstrap(conf, rowType, parallelism, dataStream, context.isBounded(), overwrite);
-      // write pipeline
-      pipeline = Pipelines.hoodieStreamWrite(conf, parallelism, hoodieRecordDataStream);
+      if (OptionsResolver.isBucketIndexTable(conf)) {
+        if (!OptionsResolver.isMorTable(conf)) {
+          throw new HoodieNotSupportedException("Bucket index only support MOR table type.");
```

Comment:
We can move the whole `if ... else ...` code block into the ` Pipelines.hoodieStreamWrite` function.

Reply:
sure